python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -528,3 +534,51 @@ def _add_prefixes(names, prefixes):
         for i, (prefix, col_name) in enumerate(zip(prefixes, names)):
             prefixed_names[i] = f"{prefix}_{col_name}"
     return prefixed_names
+
+
+def _align_by_and_df(obj, by, how="inner"):
+    """
+    Returns a pair of dataframes and a list may be containing
+    combination of column names and Series  which are intersected
+    as per their indices.
+    """
+    new_obj = None
+    new_by = []
+    if not isinstance(by, (list, tuple)):
+        by = [by]
+
+    series_count = 0
+    for i in by:
+        if not is_scalar(i):
+            sr = i
+            if not isinstance(i, cudf.Series):
+                sr = cudf.Series(i)
+            if new_obj is None:
+                new_obj = sr.to_frame(series_count)
+            else:
+                new_obj = new_obj.join(
+                    sr.to_frame(series_count), how="inner", sort="True"
+                )
+            series_count += 1
+
+    series_count = 0
+    if new_obj is not None:
+        new_obj = new_obj.join(obj, how=how, sort="True")
+        columns = new_obj.columns
+        for i in by:
+            if isinstance(i, cudf.Series):
```

Comment:
As a matter of style, `i` is typically used for integer indices, how about `by_col` instead?